File: src/main/java/com/mrkirby153/botcore/spring/EventHandler.java
package com.mrkirby153.botcore.spring;

import net.dv8tion.jda.api.events.GenericEvent;
import net.dv8tion.jda.api.hooks.EventListener;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.ApplicationEventPublisher;

import javax.annotation.Nonnull;

/**
 * Event handler to dispatch JDA events into the Spring event publisher
 */
class EventHandler implements EventListener {

    private static final Logger log = LoggerFactory.getLogger(EventHandler.class);

    private final ApplicationEventPublisher eventPublisher;

    /**
     * Constructs a new event handler
     *
     * @param eventPublisher The publisher to publish events on
     */
    public EventHandler(ApplicationEventPublisher eventPublisher) {
        this.eventPublisher = eventPublisher;
    }

    @Override
    public void onEvent(@Nonnull GenericEvent event) {
        log.trace("Dispatching event {} to spring", event.getClass());
        eventPublisher.publishEvent(event);
    }
}

File: src/main/kotlin/com/mrkirby153/botcore/spring/EventHandler.kt
package com.mrkirby153.botcore.spring

import com.mrkirby153.botcore.utils.SLF4J
import net.dv8tion.jda.api.events.GenericEvent
import net.dv8tion.jda.api.hooks.EventListener
import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationEventPublisher

/**
 * Event handler to dispatch JDA events into the Spring event publisher
 */
class EventHandler
/**
 * Constructs a new event handler
 *
 * @param eventPublisher The publisher to publish events on
 */(private val eventPublisher: ApplicationEventPublisher) : EventListener {

    private val log by SLF4J
    override fun onEvent(event: GenericEvent) {
        log.trace("Dispatching event {} to spring", event.javaClass)
        eventPublisher.publishEvent(event)
    }
}